src/main/java/dev/slimevr/vr/processor/skeleton/LegTweaks.java

```diff
@@ -0,0 +1,858 @@
+	// be if it is move it back down to the floor
+	// note: very small corrections are all that is allowed to prevent momemnt
+	// that looks unrealistic
+	private void correctFloat() {
+		boolean correctLeft = true;
+		boolean correctRight = true;
+		if (bufferHead.getLeftLegState() == LegTweakBuffer.LOCKED) {
+			float lastPositionY = bufferHead
+				.getParent()
+				.getLeftFootPositionCorrected(null).y;
+			if (
+				Math.abs(leftFootPosition.y - lastPositionY) < FOOT_Y_DIFF_CUTOFF
+					&& leftFootPosition.y > lastPositionY
+					&& bufferHead.getLeftFootAccelerationY() < FOOT_Y_MAX_ACCELERATION
+			) {
+				leftFootPosition.y = lastPositionY;
+				correctLeft = false;
+			}
+		}
+		if (bufferHead.getRightLegState() == LegTweakBuffer.LOCKED) {
+			float lastPositionY = bufferHead
+				.getParent()
+				.getRightFootPositionCorrected(null).y;
+			if (
+				Math.abs(rightFootPosition.y - lastPositionY) < FOOT_Y_DIFF_CUTOFF
+					&& rightFootPosition.y > lastPositionY
+					&& bufferHead.getRightFootAccelerationY() < FOOT_Y_MAX_ACCELERATION
+			) {
+				rightFootPosition.y = lastPositionY;
+				correctRight = false;
+			}
+		}
+
+		// using the velocity correct the position
+		if (correctLeft) {
+			correctLeftFootTrackerY();
+		}
+
+		if (correctRight) {
+			correctRightFootTrackerY();
+		}
+	}
+
+	private void correctLeftFootTrackerY() {
+		Vector3f temp;
+		Vector3f leftFootDif = leftFootPosition
+			.subtract(bufferHead.getParent().getLeftFootPositionCorrected(null));
+		if (Math.abs(leftFootDif.y) > NEARLY_ZERO) {
+			temp = bufferHead.getParent().getLeftFootPositionCorrected(null);
+			temp = temp
+				.subtract(
+					bufferHead
+						.getParent()
+						.getLeftFootPosition(null)
+						.subtract(bufferHead.getLeftFootPosition(null))
+				);
+			float leftFloor = (floorLevel + (MAX_DYNAMIC_DISPLACEMENT * getLeftFootOffset()))
+				- currentDisengagementOffset;
+			if (
+				temp.y
+					< leftFloor
+			) {
+				leftFootPosition.y = leftFloor;
+			} else {
+				leftFootPosition.y = temp.y;
+			}
+
+			Vector3f velocity = bufferHead.getLeftFootVelocity(null);
+			if (velocity.y * leftFootDif.y > 0) {
+				leftFootPosition.y += velocity.y * FOOT_Y_CORRECTION_WEIGHT;
+			} else {
+				leftFootPosition.y -= velocity.y * FOOT_Y_CORRECTION_WEIGHT;
+			}
+			// check for overshoot and correct if necessary
+			if (
+				checkOverShoot(
+					this.bufferHead.getLeftFootPosition(null).y,
+					this.bufferHead.getParent().getLeftFootPositionCorrected(null).y,
+					leftFootPosition.y
+				)
+			) {
+				leftFootPosition.y = bufferHead.getLeftFootPosition(null).y;
+			}
+		}
+	}
+
+	private void correctRightFootTrackerY() {
+		Vector3f temp;
+		Vector3f rightFootDif = rightFootPosition
+			.subtract(bufferHead.getParent().getRightFootPositionCorrected(null));
+		if (Math.abs(rightFootDif.y) > NEARLY_ZERO) {
+			temp = bufferHead.getParent().getRightFootPositionCorrected(null);
+			temp = temp
```

Comment:
```suggestion
			temp = bufferHead.getParent().getRightFootPositionCorrected(null)
```

please reformat after this ^^